crates/net/network/src/session/active.rs

```diff
@@ -84,6 +84,9 @@ pub(crate) struct ActiveSession {
     pub(crate) internal_request_tx: Fuse<ReceiverStream<PeerRequest>>,
     /// All requests sent to the remote peer we're waiting on a response
     pub(crate) inflight_requests: FnvHashMap<u64, InflightRequest>,
+    /// Counting semaphore notifies [`crate::transactions::TransactionsManager`] of total inflight
+    /// requests.
+    pub(crate) inflight_requests_semaphore_tx: watch::Sender<usize>,
```

Comment:
just a q: if there a reason for implementing a counting semaphore this way instead of just using `Semaphore` / `PollSemaphore`?

Reply:
Thanks for the tip, haven't used that type before :)